data_collection_scripts/collect_block_data.py

```diff
@@ -1,39 +1,40 @@
 """
     This script can be used to run queries on BigQuery for any number of blockchains, and save the results in the
     raw_block_data directory of the project.
-    The relevant queries must be stored in a file named 'queries.yaml' in the root directory of the project.
+    The relevant queries must be stored in a file named 'queries.yaml' in the `data_collection_scripts` directory of
+    the project.
 
     Attention! Before running this script, you need to generate service account credentials from Google, as described
     here (https://developers.google.com/workspace/guides/create-credentials#service-account) and save your key in the
-    root directory of the project under the name 'google-service-account-key.json'
+    `data_collection_scripts` directory of the project under the name 'google-service-account-key.json'
 """
 import consensus_decentralization.helper as hlp
 import google.cloud.bigquery as bq
 import json
 import argparse
 import logging
 from yaml import safe_load
+from datetime import datetime
 
 from consensus_decentralization.helper import ROOT_DIR, RAW_DATA_DIR
 
 
-def collect_data(ledgers, force_query):
+def collect_data(ledgers, from_block, to_date):
     if not RAW_DATA_DIR.is_dir():
         RAW_DATA_DIR.mkdir()
 
-    with open(ROOT_DIR / "queries.yaml") as f:
+    data_collection_dir = ROOT_DIR / "data_collection_scripts"
+
+    with open(data_collection_dir / "queries.yaml") as f:
         queries = safe_load(f)
 
-    client = bq.Client.from_service_account_json(json_credentials_path=ROOT_DIR / "google-service-account-key.json")
+    client = bq.Client.from_service_account_json(json_credentials_path=data_collection_dir / "google-service-account-key.json")
 
     for ledger in ledgers:
         file = RAW_DATA_DIR / f'{ledger}_raw_data.json'
-        if not force_query and file.is_file():
-            logging.info(f'{ledger} data already exists locally. '
-                         f'For querying {ledger} anyway please run the script using the flag --force-query')
-            continue
         logging.info(f"Querying {ledger}..")
-        query = (queries[ledger])
+
+        query = (queries[ledger]).replace("{{block_number}}", str(from_block[ledger]) if from_block[ledger] else "-1").replace("{{timestamp}}", f"'{to_date}'")

```

Comment:
I think this ```f"'{to_date}'"``` can just be ```to_date```.

Reply:
That's how I had it at first but there was an issue with the lack of quotation marks

Reply:
What issue?

Reply:
![image](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/assets/20790332/44d5cd6b-297f-490f-af71-2d89920b8a04)